libs/utils/WordPressUtils/src/main/java/org/wordpress/android/util/FileUtils.java

```diff
@@ -58,4 +73,127 @@ public static String getFileNameFromPath(String filePath) {
         }
         return filename;
     }
+
+    /**
+     * This solution is based on https://stackoverflow.com/a/53021624
+     * In certain cases we cannot load a file from the disk so we have to cache it instead using streams.
+     * The helper methods are copied from - https://github.com/coltoscosmin/FileUtils/blob/master/FileUtils.java
+     *
+     * @param context The context.
+     * @param uri The Uri to query.
+     */
+    static String cacheFile(Context context, Uri uri) {
+        String fileName = getFileName(context, uri);
+        File cacheDir = getDocumentCacheDir(context);
+        File file = generateFileName(fileName, cacheDir);
+        String destinationPath = null;
+        if (file != null) {
+            destinationPath = file.getAbsolutePath();
+            saveFileFromUri(context, uri, destinationPath);
+        }
+        return destinationPath;
+    }
+
+    private static String getFileName(Context context, Uri uri) {
+        String mimeType = context.getContentResolver().getType(uri);
+        String filename = null;
+
+        if (mimeType == null) {
+            String path = getPath(context, uri);
+            if (path == null) {
+                filename = getName(uri.toString());
+            } else {
+                File file = new File(path);
+                filename = file.getName();
+            }
+        } else {
+            Cursor returnCursor = context.getContentResolver().query(uri, null,
+                    null, null, null);
+            if (returnCursor != null) {
+                int nameIndex = returnCursor.getColumnIndex(OpenableColumns.DISPLAY_NAME);
+                returnCursor.moveToFirst();
+                filename = returnCursor.getString(nameIndex);
+                returnCursor.close();
+            }
+        }
+
+        return filename;
+    }
+
+    private static String getName(String filename) {
+        if (filename == null) {
+            return null;
+        }
+        int index = filename.lastIndexOf('/');
+        return filename.substring(index + 1);
+    }
+
+    private static File getDocumentCacheDir(@NonNull Context context) {
+        File dir = new File(context.getCacheDir(), DOCUMENTS_DIR);
+        if (!dir.exists()) {
+            dir.mkdirs();
+        }
+        return dir;
+    }
+
+    @Nullable
+    private static File generateFileName(@Nullable String name, File directory) {
```

Comment:
Wondering it we can re-use the already available routine [MediaUtils.getUniqueCacheFileForName ](https://github.com/wordpress-mobile/WordPress-Android/blob/bb8f7ee2d4d02c960b6e76beb14f6bedf4d1fc68/libs/utils/WordPressUtils/src/main/java/org/wordpress/android/util/MediaUtils.java#L241) ?

Reply:
I think we can! good point 👍